Removed manifest-tool dependency from pause image

AnitaNayak opened the pull request:
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Removed manifest-tool dependency from pause Image.

**Which issue(s) this PR fixes**:
To fix  https://github.com/kubernetes/kubernetes/issues/69358

**Special notes for your reviewer**:
**Does this PR introduce a user-facing change?**:
NONE

AnitaNayak committed:
Removed manifest-tool dependency from pause image

AnitaNayak commented:
Test results:
```
staging-k8s.gcr.io/pause-s390x                               3.1                            61c8b1795d39        41 seconds ago      537kB
staging-k8s.gcr.io/pause-ppc64le                             3.1                            f753b1b2e56f        44 seconds ago      731kB
staging-k8s.gcr.io/pause-arm64                               3.1                            c1c43d6a8b64        47 seconds ago      484kB
staging-k8s.gcr.io/pause-arm                                 3.1                            4dc85d77df0f        50 seconds ago      325kB
staging-k8s.gcr.io/pause-amd64                               3.1                            00a2c3a7941b        53 seconds ago      683kB
```
Please let me know if any other test results needs to be updated.
Version is still 3.1. Let me know if this needs to be bumped.

neolit123 commented:
/assign @mkumatag 
please take a look. thanks :)

neolit123 commented:
/release-note-none
/ok-to-test
/priority important-longterm
/kind cleanup

neolit123 commented:
/sig release

fejta reviewed (commented):
/approve
/ok-to-test

mkumatag commented:
/hold

mkumatag commented:
@AnitaNayak the ask https://github.com/kubernetes/kubernetes/issues/69358 is to use `docker manifest` command instead of manifest-tool command, please refactor the code to use docker command instead, you can refer https://github.com/kubernetes/kubernetes/blob/master/test/images/image-util.sh#L134:L138 to understand how we are using `docker manifest` command.

AnitaNayak commented:
Thank you for reviewing. Will update it soon.

AnitaNayak commented:
Hi @mkumatag , made the changes to use docker manifest. Could you please review the latest changes.

mkumatag commented on hack/make-rules/Makefile.manifest:
```diff
@@ -1,30 +0,0 @@
-# Copyright 2018 The Kubernetes Authors.
```
I think it is still been refered in https://github.com/kubernetes/kubernetes/blob/master/test/images/Makefile#L15, need to check whether is it really been used anywhere in the `test/images` or just a stale entry!

AnitaNayak replied:
Its a miss. Thought of deleting, but missed. Will update it. Thank you.

mkumatag commented on build/pause/Makefile:
```diff
@@ -94,6 +95,12 @@ push: .push-$(ARCH)
 	touch $@
 
+push-manifest: .manifest-$(ARCH)
+.manifest-$(ARCH): 
```
This `.manifest-$(ARCH)` is not really required!

AnitaNayak replied:
.manifest-$(ARCH) not required. target push-manifest is good enough. Thank you. Will update and push.

AnitaNayak replied:
Made a dependency on `.container-$(ARCH)` and pushed the changes.

AnitaNayak commented:
Updated as per review comment. Please review.
1. Restored file `hack/make-rules/Makefile.manifest`
2. Updated push-manifest target as follow (made a dependency `.container-$(ARCH)` for .manifest-$(ARCH))
```
push-manifest: .manifest-$(ARCH)
.manifest-$(ARCH): .container-$(ARCH)
```

AnitaNayak commented:
Hi @mkumatag , could you please review the latest changes. Thank you.

mkumatag commented on build/pause/Makefile:
```diff
@@ -94,6 +95,11 @@ push: .push-$(ARCH)
 
+push-manifest:
+	docker manifest create --amend $(IMAGE):$(TAG) $(foreach ALL_ARCH,$(ALL_ARCH),$(IMAGE)-$(ALL_ARCH):$(TAG))
+	docker manifest annotate --arch $(ARCH) $(IMAGE):$(TAG) $(IMAGE_WITH_ARCH):$(TAG)
```
Just one arch annotate is not good enough, you need to loop through all possible archs annotation. Refer code here:

https://github.com/kubernetes/kubernetes/blob/master/test/images/image-util.sh#L135:L137

mkumatag commented:
This is already taken care via https://github.com/kubernetes/kubernetes/pull/88139

/close
